src/crush/builder.c

```diff
@@ -82,6 +82,7 @@ int crush_add_rule(struct crush_map *map, struct crush_rule *rule, int ruleno)
 		int oldsize;
 		void *_realloc = NULL;
 		if (map->max_rules +1 > CRUSH_MAX_RULES)
+			free(rule);
```

Comment:
i'd suggest `free(rule)` in the caller site of this function. because it's "closer" to where `rule` is allocated. what do you think?